runtime/j9vm/java11vmi.c

```diff
@@ -710,18 +711,79 @@ allowReadAccessToModule(J9VMThread * currentThread, J9Module * fromModule, J9Mod
  * @return If successful, returns a java.lang.reflect.Module object. Otherwise, returns NULL.
  */
 jobject JNICALL
+#if JAVA_SPEC_VERSION >= 15
+JVM_DefineModule(JNIEnv * env, jobject module, jboolean isOpen, jstring version, jstring location, jobjectArray packageArray)
+#else
 JVM_DefineModule(JNIEnv * env, jobject module, jboolean isOpen, jstring version, jstring location, const char* const* packages, jsize numPackages)
+#endif /* JAVA_SPEC_VERSION >= 15 */
 {
 	J9VMThread * const currentThread = (J9VMThread*)env;
 	J9JavaVM * vm = currentThread->javaVM;
 	J9InternalVMFunctions const * const vmFuncs = vm->internalVMFunctions;
+#if JAVA_SPEC_VERSION >= 15
+	BOOLEAN oom = FALSE;
+	BOOLEAN nullPackage = FALSE;
+	jsize numPackages = (NULL == packageArray) ? 0 : (*env)->GetArrayLength(env, packageArray);
+	PORT_ACCESS_FROM_ENV(env);
+	UDATA packagesNumBytes = sizeof(char*) * numPackages;
+	const char** packages = NULL;
+#endif /* JAVA_SPEC_VERSION >= 15 */
 
 	vmFuncs->internalEnterVMFromJNI(currentThread);
 #if defined(CALL_BUNDLED_FUNCTIONS_DIRECTLY)
 	omrthread_monitor_enter(vm->classLoaderModuleAndLocationMutex);
 #else
 	f_monitorEnter(vm->classLoaderModuleAndLocationMutex);
 #endif /* defined(CALL_BUNDLED_FUNCTIONS_DIRECTLY) */
+
+#if JAVA_SPEC_VERSION >= 15
+	if ((*env)->ExceptionCheck(env)) {
+		goto done;
+	}
+	packages = (const char**)j9mem_allocate_memory(packagesNumBytes, OMRMEM_CATEGORY_VM);
+	if (NULL != packages) {
+		jsize pkgIndex = 0;
+		memset((void *)packages, 0, packagesNumBytes);
+		for (pkgIndex = 0; pkgIndex < numPackages; pkgIndex++) {
+			jobject packageObj = (*env)->GetObjectArrayElement(env, packageArray, pkgIndex);
+			if ((*env)->ExceptionCheck(env)) {
+				(*env)->DeleteLocalRef(env, packageObj);
```

Comment:
See https://github.com/eclipse/openj9/pull/9292#discussion_r412239235: using JNI after calling `internalEnterVMFromJNI()` is not legal.

Reply:
Removed those JNI calls.